src/types/Stream.d.ts

```diff
@@ -0,0 +1,50 @@
+import Client from './Client'
+import { ExprArg } from './query'
+import { values } from './values'
+
+export interface Subscription {
+  on: <T extends keyof SubscriptionEventHandlers>(
+    type: T,
+    callback: SubscriptionEventHandlers[T]
+  ) => this
+  start: () => this
+  close: () => void
+}
+
+type SubscriptionEventHandlers = {
+  start: (type: 'start', txn: number, event: number) => void
+  error: (
+    type: 'error',
+    txn: number,
+    event: {
+      code: string
+      description: string
+    }
+  ) => void
+  version: (
+    type: 'version',
+    txn: number,
+    event: {
+      action: 'create' | 'update' | 'delete'
+      document: object
+      diff: object
+      prev: object
+    }
+  ) => void
+  history_rewrite: (
+    type: 'history_rewrite',
+    txn: number,
+    event: {
+      action: 'history_rewrite'
+      document: object
+    }
+  ) => void
+  snapshot: (
+    type: 'snapshot',
+    txn: number,
+    event: {
+      action: 'snapshot'
```

Comment:
A snapshot has no action. The event itself is the document.

Reply:
@erickpintor In that case, would `event: object` be the better definition? Or is there an `event.document` field that should be an `object`?

Reply:
`event: object` should it. There's no `document` field in it.